manager/orchestrator/global/global_test.go

```diff
@@ -221,28 +258,72 @@ func TestDeleteService(t *testing.T) {
 	assert.Equal(t, observedTask.NodeID, "nodeid1")
 }
 
-func TestRemoveTask(t *testing.T) {
+func TestTaskFailure(t *testing.T) {
 	store := store.NewMemoryStore(nil)
 	assert.NotNil(t, store)
 	defer store.Close()
 
 	watch, cancel := state.Watch(store.WatchQueue() /*api.EventCreateTask{}, api.EventUpdateTask{}*/)
 	defer cancel()
 
-	observedTask1 := SetupCluster(t, store, watch)
+	// first, try a "restart on any" policy
+	orchestrator := setup(t, store, watch)
+	defer orchestrator.Stop()
+
+	observedTask1 := testutils.WatchTaskCreate(t, watch)
 
 	assert.Equal(t, observedTask1.Status.State, api.TaskStateNew)
 	assert.Equal(t, observedTask1.ServiceAnnotations.Name, "name1")
 	assert.Equal(t, observedTask1.NodeID, "nodeid1")
 
-	// delete the task
-	deleteTask(t, store, observedTask1)
+	failTask(t, store, observedTask1)
+
+	testutils.WatchShutdownTask(t, watch)
 
 	// the task should be recreated
 	observedTask2 := testutils.WatchTaskCreate(t, watch)
 	assert.Equal(t, observedTask2.Status.State, api.TaskStateNew)
 	assert.Equal(t, observedTask2.ServiceAnnotations.Name, "name1")
 	assert.Equal(t, observedTask2.NodeID, "nodeid1")
+	testutils.Expect(t, watch, state.EventCommit{})
+	testutils.Expect(t, watch, api.EventUpdateTask{}) // ready->running
+	testutils.Expect(t, watch, state.EventCommit{})
+
```

Comment:
Would it make sense to also delete the task here, and assert that it doesn't get re-created, since that behavior has also changed?